buildbot/cmd/buildbot/rotation.go

```diff
@@ -0,0 +1,83 @@
+package main
+
+import (
+	"encoding/csv"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"time"
+
+	"k8s.io/kubernetes/pkg/kubelet/kubeletconfig/util/log"
+)
+
+// The Rotation object which knows how to dynamically find the correct build cop from
+// the file getter f it is initialized with.
+type Rotation struct {
+	f GetFile
+}
+
+// GetFile is the signature of a function that knows how to retrieve the bytes from a file
+type GetFile func() (io.ReadCloser, error)
+
+// NewRotationFromFile returns a new Rotation object which will read from file.
+func NewRotationFromFile(file string) Rotation {
+	return Rotation{f: func() (io.ReadCloser, error) {
+		f, err := os.Open(file)
+		if err != nil {
+			return nil, fmt.Errorf("Could not open file %s: %v", file, err)
+		}
+		return f, nil
+	}}
+}
+
+// NewRotationFromURL returns a new Rotation object which will read from url.
+func NewRotationFromURL(url string) Rotation {
+	return Rotation{f: func() (io.ReadCloser, error) {
+		resp, err := http.Get(url)
+		if err != nil {
+			return nil, fmt.Errorf("Could not open url %s: %v", url, err)
+		}
+		return resp.Body, nil
+	}}
+}
+
+// GetBuildCop returns the name of the build cop for the requested time
+func (r Rotation) GetBuildCop(t time.Time) string {
+	tf := t.Format("2006-01-02") // Mon Jan 2 15:04:05 MST 2006
+	f, err := r.f()
+	if err != nil {
+		log.Errorf("Could not read from build cop rotation: %v", err)
```

Comment:
Any reason to use `log` here instead of `fmt` ?
Also, we should *most likely* use the built-in log instead of `k8s.io/kubernetes/pkg/kubelet/kubeletconfig/util/log"`, as.. bringing k8s.io/kubernetes dependency just for logs is a high coste :angel:

Reply:
oh dear i did NOT mean to bring that in! that explains the go.mod change 🤦‍♀

Reply:
i meant to use the built in `log` lib, which i thought made a bit more sense than `fmt` but im open to either!

Reply:
ah lol.. I'll wait for your follow-up :stuck_out_tongue_closed_eyes: